### PR TITLE
editoast: remove `list` impl from `LightRollingStockModel`

### PR DESCRIPTION
**Pull request by hamz2a**
1. Remove `list` impl from `LightRollingStockModel`
2. Remove `with_liveries` impl from `LightRollingStockModel` and `RollingStockModel`
3. Remove `RollingStockLiveryMetadataModel`
4. Replace `PaginatedResponse<LightRollingStockWithLiveries>` by `LightRollingStockWithLiveriesCountList` in `views::light_rolling_stocks`
5. Rename `rolling_stock_id` by `light_rolling_stock_id` in `views::light_rolling_stock`
6. Refactor a little bit the code

### CODE REVIEW

File: editoast/src/modelsv2/light_rolling_stock.rs [hunk 1]
```diff
@@ -65,97 +59,34 @@ pub struct LightRollingStockModel {
 }
 
 impl LightRollingStockModel {
-    pub async fn with_liveries(
-        self,
+    pub async fn liveries(
+        &self,
         conn: &mut DbConnection,
-    ) -> Result<LightRollingStockWithLiveriesModel> {
+    ) -> Result<Vec<RollingStockLiveryMetadata>> {
```
Comment (leovalais) [inline editoast/src/modelsv2/light_rolling_stock.rs]:
```suggestion
    ) -> Result<Vec<RollingStockLiveryModel>> {
```

1. We have two very similar structs for liveries in `modelsv2`: `RollingStockLiveryModel` and `RollingStockLiveryMetadataModel`. I suggest getting rid of the latter.
2. We're in the `modelsv2` module, implementing an operation on a model that fetches another model. There's no need to convert into a schema this early.

File: editoast/src/modelsv2/light_rolling_stock.rs [hunk 2]
```diff
@@ -65,97 +59,34 @@ pub struct LightRollingStockModel {
...
-        per_page: i64,
-    ) -> Result<PaginatedResponse<LightRollingStockWithLiveriesModel>> {
-        let light_rolling_stocks = sql_query("SELECT * FROM rolling_stock ORDER BY id")
-            .paginate(page, per_page)
-            .load_and_count::<Row<LightRollingStockModel>>(conn)
-            .await?;
-
-        let lrs_results: Vec<LightRollingStockModel> = light_rolling_stocks
-            .results
+            .await?
```
Comment (leovalais) [inline editoast/src/modelsv2/light_rolling_stock.rs]: With the liveries model deriving `ModelV2`, you can use `::list` to do that.

File: editoast/src/views/light_rolling_stocks.rs [hunk 3]
```diff
@@ -40,6 +46,35 @@ editoast_common::schemas! {
     LightRollingStockWithLiveries,
 }
 
+#[derive(Debug, Serialize, Deserialize, ToSchema)]
```
Comment (leovalais) [inline editoast/src/views/light_rolling_stocks.rs]: Does this need to be `Deserialize`?

File: editoast/openapi.yaml
Comment (leovalais) [inline editoast/openapi.yaml:1]: The old paginated response should be deleted.

File: editoast/src/views/light_rolling_stocks.rs [hunk 4]
```diff
@@ -81,19 +123,23 @@ async fn list(
...
     .await?;
-    let rollig_stock_with_liveries: LightRollingStockWithLiveries =
-        rolling_stock.with_liveries(conn).await?.into();
-    Ok(Json(rollig_stock_with_liveries))
+    let light_rolling_stock_with_liveries = LightRollingStockWithLiveries::try_fetch(
+        db_pool.get().await?.deref_mut(),
+        light_rolling_stock,
+    )
+    .await?;
+    Ok(Json(light_rolling_stock_with_liveries))
```
Comment (leovalais) [inline editoast/src/views/light_rolling_stocks.rs]: The renaming feels a bit unrelated to the topic of the PR. I'm fine with this, but better make it clear in the description of the PR (after looking back at it I suppose this falls back in the `Refactor a little bit the code` item).

File: editoast/src/views/rolling_stocks/light_rolling_stock.rs [hunk 5]
```diff
@@ -42,11 +41,38 @@ pub struct LightRollingStock {
     #[schema(required)]
     pub power_restrictions: HashMap<String, String>,
     pub energy_sources: Vec<EnergySource>,
-    #[serde(skip)]
```
Comment (leovalais) [inline editoast/src/views/rolling_stocks/light_rolling_stock.rs:45]: Why was this removed?

**Review by leovalais — CHANGES_REQUESTED** [review]
Our rolling stock implementation really need to be re-architectured, thanks for starting it with this PR.

File: editoast/src/views/light_rolling_stocks.rs (hunk 3 above)
Comment (hamz2a) [reply editoast/src/views/light_rolling_stocks.rs]: only for tests, I will use this `#[cfg_attr(test, derive(Deserialize))]`

File: editoast/src/views/light_rolling_stocks.rs (hunk 4 above)
Comment (hamz2a) [reply editoast/src/views/light_rolling_stocks.rs]: I added this specific renaming in the description.

File: editoast/src/views/rolling_stocks/light_rolling_stock.rs (hunk 5 above)
Comment (hamz2a) [reply editoast/src/views/rolling_stocks/light_rolling_stock.rs:45]: I had a clippy error like `clippy error: field 'version' is never read`

Comment (leovalais) [reply editoast/src/views/rolling_stocks/light_rolling_stock.rs:45]: In this case removing the field seems more fitting to me.

Comment (hamz2a) [reply editoast/src/views/rolling_stocks/light_rolling_stock.rs:45]: Done.

File: editoast/src/views/light_rolling_stocks.rs [hunk 6]
```diff
@@ -25,44 +35,85 @@ crate::routes! {
...
+        let light_rolling_stock_id = light_rolling_stock.id;
+        let liveries = RollingStockLiveryModel::list(
+            conn,
+            SelectionSettings::new().filter(move || {
+                RollingStockLiveryModel::ROLLING_STOCK_ID.eq(light_rolling_stock_id)
+            }),
+        )
+        .await?
+        .into_iter()
+        .map(Into::into)
```
Comment (leovalais) [inline editoast/src/views/light_rolling_stocks.rs]:
```suggestion
        .map_into()
```

**Review by leovalais — APPROVED** [review]
LGTM, tested, great work!

**Review by SharglutDev — APPROVED** [review]
Lgtm (not tested)
